### DISCUSSION TITLE
Add a sequential response toggle

### DISCUSSION THREAD

**Discussion by davidvieiratrustly** [mockoon/mockoon#458] [Ideas] [answered]
Add an option, next to the "Random response" toggle, to be able to make a sequential response for the same endpoint:

Example of one endpoint:

**GET /device/ready**
{"status": "preparing"} <- response 1
**GET /device/ready**
{"status": "ready_to_pair"} <- response 2
**GET /device/ready**
{"status": "ready_to_pair"} <- response 3
**GET /device/ready**
{"status": "paired", "pair_code": "xyz"} <- response 4

After the last response, they loop to the first one:
**GET /device/ready**
{"status": "preparing"} <- response 1
...

**Comment by 255kb** [comment, marked as answer]
Hi @davidvieiratrustly this is a good idea and have been requested before. 
I guess, instead of a new button, this could be done in two ways:
- a new helper `requestNumber` that could allow modifying the body for each call
- a new type of rules that would allow serving different responses depending on the "index" of the call. 

Could you  please open a new issue so we can properly track this feature request? 

Thank you

**Comment by davidvieiratrustly** [comment]
Hi @255kb, thanks for the reply!

I like the new rule idea better because it would be possible to make a default response and others based on this index. I'm a little bit afraid about the discoverability of this functionality, but it's probably a minor issue.

I don't like the requestNumber idea, because I would not be able to change the response status code for a particular case.

I'll open the issue now.

Thank you again

**Reply by davidvieiratrustly** [reply]
Issue https://github.com/mockoon/mockoon/issues/459

**Reply by 255kb** [reply]
Great thank you @davidvieiratrustly 
It's true that discoverability is important, but not cluttering the UI is also important. It's always hard to choose 🤔 
Also I agree concerning the body helper, it's not the best solution.
